src/main/kotlin/brs/db/sql/SqlIndirectIncomingStore.kt

```diff
@@ -33,30 +32,33 @@ internal class SqlIndirectIncomingStore(private val dp: DependencyProvider) : In
                 )
             }
 
-            private fun getQuery(ctx: DSLContext, indirectIncoming: IndirectIncoming): Query {
-                val record = IndirectIncomingRecord()
-                record.accountId = indirectIncoming.accountId
-                record.transactionId = indirectIncoming.transactionId
-                record.height = indirectIncoming.height
-                return ctx.upsert(record, INDIRECT_INCOMING.ACCOUNT_ID, INDIRECT_INCOMING.TRANSACTION_ID)
-            }
-
             override fun save(ctx: DSLContext, entity: IndirectIncoming) {
-                getQuery(ctx, entity).execute()
-            }
-
-            override fun save(ctx: DSLContext, entities: Array<IndirectIncoming>) {
-                val queries = mutableListOf<Query>()
-                for (indirectIncoming in entities) {
-                    queries.add(getQuery(ctx, indirectIncoming))
-                }
-                ctx.batch(queries).execute()
+                val record = IndirectIncomingRecord()
+                record.accountId = entity.accountId
+                record.transactionId = entity.transactionId
+                record.height = entity.height
+                ctx.upsert(record, INDIRECT_INCOMING.ACCOUNT_ID, INDIRECT_INCOMING.TRANSACTION_ID)
             }
         }
     }
 
     override fun addIndirectIncomings(indirectIncomings: Collection<IndirectIncoming>) {
-        dp.db.useDslContext { ctx -> indirectIncomingTable.save(ctx, indirectIncomings.toTypedArray()) }
+        dp.db.useDslContext { ctx ->
```

Comment:
Please could this be moved to the function you removed in the for consistency with other `Store` classes?